spec/classes/init_spec.rb

```diff
@@ -1,6 +1,12 @@
 require 'spec_helper'
 describe 'sudo' do
   context 'with class default options' do
+    let :facts do
+      {
+        :architecture => 'x86_64',
+      }
+    end
```

Comment:
Wouldn't it make sense to define default facts for the spec tests and override them only when needed ?
See https://github.com/ghoneycutt/puppet-module-nrpe/commit/bf58520ef7fcab0a1f57da7f65e81c2feddaf435 for an example.